tremor-pipeline/src/lib.rs

```diff
@@ -417,12 +418,20 @@ impl EventId {
     /// track the min and max of the given `event_id`
     /// and also include all event ids `event_id` was tracking
     pub fn track(&mut self, other: &EventId) {
-        self.track_ids(
-            other.source_id,
-            other.stream_id,
-            other.pull_id,
-            other.pull_id,
-        );
+        // if self.source_id and self.stream_id are the same, calculate the max from the given pull_id and the one in other
+        // and insert it with binary search
+        // this makes searching the tracked id easier
+        // as we only need to look at the tracked_pull_ids then
+        let (min, max) =
+            if self.source_id() == other.source_id() && self.stream_id() == other.stream_id() {
```

Comment:
since this is happening a few times, would it make sense to have a `self.same_stream(other: &Self)` function?

Reply:
good point! :)